sys/net/include/fib/fib.h

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2014  Martin Landsmann <Martin.Landsmann@HAW-Hamburg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_fib Forwarding Information Base (FIB)
+ * @brief       FIB implementation
+ * @ingroup     net
+ *
+ * @{
+ *
+ * @file        fib.h
+ * @brief       Types and functions for FIB
+ * @author      Martin Landsmann
+ */
+
+#ifndef _FIB_H
+#define _FIB_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Reactive Routing Protocol (RRP) message content to request/reply route discovery
+ */
+typedef struct rrp_address_msg_t {
+    uint8_t *address;      /**< The pointer to the address */
+    uint8_t address_size;  /**< The address size */
+    kernel_pid_t iface_id; /**< The interface used for the next-hop reply.
+                                If it is KERNEL_PID_UNDEF, no next-hop has been found. */
+    uint32_t lifetime;     /**< The lifetime of this next-hop */
+} rrp_address_msg_t;
+
+#define FIB_MSG_RRP_SIGNAL (0x99)              /**< message type for RRP notifications         */
```

Comment:
But what would be the procedure if we have say 3 RRPs available, which iface/handler ID should be then used?

Simple scenario: 
3 transceivers with each used by an individual RRP and the FIB is empty. Potentially every of the 3 RRP may provide a route to the destination.